### DISCUSSION TITLE
Image not sharp when resizing from very large to small thumbnail

### DISCUSSION THREAD

**Discussion by gczh** [janko/image_processing#111] [Q&A]
Hey there!

I'm currently trying to convert a fairly large image into a thumbnail. 

The original image dimensions are [2880, 16446]

The target output sizes are roughly [359, 330] (I know it's a large difference!)

Here's the code for it:

```ruby
thumb = ImageProcessing::Vips
      .source(file)
      .resize_to_limit(target_width, target_height)
      .crop(0, 0, width.to_i, height.to_i)
      .convert("webp")
      .saver(quality: 100)
      .call
```

When converting to thumbnail, the image looks really pixelated

![telegram-cloud-photo-size-5-6212838294845241129-y](https://github.com/janko/image_processing/assets/1908934/4ffa6869-4a01-4455-8543-2f69a8b64f7f)

Is there any way to resize and crop it without losing it's clarity?

**Edit:** 
- Probably not the best way to do it but what i did instead is resize to 2x the thumbnail size to 718x660). When rendered in actual thumbnail dimensions (359x330), it'll look sharp.
- Would really want to learn if there's a better way around this!

Appreciate any help here! 🙏

**Comment by janko** [comment]
This would probably better asked at [libvips discussions](https://github.com/libvips/libvips). The maintainer is very responsive, and he's also the maintainer of ruby-vips, so as long as you provide the equivalent processing code using this library directly (instead of ImageProcessing), I'm sure he'll be able to give you some pointers.

I believe your code translates to:

```rb
require "vips"
require "tempfile"

SHARPEN_MASK = ::Vips::Image.new_from_array [[-1, -1, -1],
                                             [-1, 32, -1],
                                             [-1, -1, -1]], 24

output = Tempfile.new ["result", ".webp"]

Vips::Image.new_from_file(file.path)
  .autorot
  .thumbnail_image(target_width, height: target_height, no_rotate: true)
  .conv(SHARPEN_MASK, precision: :integer)
  .crop(0, 0, width.to_i, height.to_i)
  .write_to_file(output.path, quality: 100)
```

Alternatively, you could try disabling ImageProcessing's default sharpening, maybe it's making the output worse:

```rb
.resize_to_limit(target_width, target_height, sharpen: false)
```

Or maybe try enabling lossless compression instead of setting quality:

```rb
.saver(lossless: true)
```

**Reply by gczh** [reply]
Thanks @janko! Appreciate all your help and amazing work with your gems!! 

I'll try your suggestions and reach out in the libvips discussion!

BTW, is there no way I could do all of the above code with `image_processing`?

I'm currently running the code I shared in a shrine-rb block

```ruby
derivation :thumbnail do |file, width, height|
# process image here
end
```

is it a good idea to use Vips::Image directly and write to a file for each variant I want to create? Wondering how you'd approach it.

I also tried the following using your alternative suggestion:

```ruby
thumb = ImageProcessing::Vips
      .source(file)
      .resize_to_limit(target_width, target_height, sharpen: false)
      .crop(0, 0, width.to_i, height.to_i)
      .convert("webp")
      .saver(lossless: true)
      .call
```

The image is still blurry 😢

**Reply by janko** [reply]
> is it a good idea to use Vips::Image directly and write to a file for each variant I want to create? Wondering how you'd approach it.

Anything you can do in ruby-vips you should generally also be able to do in ImageProcessing. This code was just a starting point for opening a discussion in libvips repo.
